File: frontend/src/types/order.ts
import type { Product } from './product';

export interface OrderItem {
  id: number;
  product_id: number;
  quantity: number;
  price_at_time: number;
  product: Product;
}

export interface Order {
  id: number;
  total: number;
  status: string;
  shipping_address: string;
  created_at: string;
  items: OrderItem[];
}

File: backend/app/schemas/order.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.product import ProductResponse


class OrderCreate(BaseModel):
    shipping_address: str


class OrderItemResponse(BaseModel):
    id: int
    product_id: int
    quantity: int
    price_at_time: float
    product: ProductResponse

    class Config:
        from_attributes = True


class OrderResponse(BaseModel):
    id: int
    total: float
    status: str
    shipping_address: str
    created_at: datetime
    items: list[OrderItemResponse]

    class Config:
        from_attributes = True


class OrderStatusUpdate(BaseModel):
    status: str
